pkg/v2/volumes.go

```diff
@@ -0,0 +1,466 @@
+// Copyright (c) 2023 Seagate Technology LLC and/or its Affiliates
+
+package mcapi
+
+import (
+	"fmt"
+	"net/http"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/Seagate/seagate-exos-x-api-go/pkg/common"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"k8s.io/klog/v2"
+)
+
+const (
+	ApiMaximumLUN         = 255
+	ApiTierAffinity       = "no-affinity"
+	ApiSuccess            = 0
+	ApiVolumeDoesNotExist = 1
+	ApiVolumeDeleted      = 2
+)
+
+// Volume : volume-view representation
+type Volume struct {
+	LUN int
+}
+
+type Volumes []Volume
+
+func (v Volumes) Len() int {
+	return len(v)
+}
+
+func (v Volumes) Swap(i, j int) {
+	v[i], v[j] = v[j], v[i]
+}
+
+func (v Volumes) Less(i, j int) bool {
+	return v[i].LUN < v[j].LUN
+}
+
+// CreateVolume : creates a volume with the given name, capacity in the given pool
+func (client *Client) CreateVolume(name, size, pool, poolType string) (*common.VolumeObject, *common.ResponseStatus, error) {
+
+	logger := klog.FromContext(client.Ctx)
+	response, httpRes, err := client.apiClient.DefaultApi.CreateVolumePoolSizeTierAffinityNameGet(client.Ctx, pool, size, ApiTierAffinity, name).Execute()
+	logger.V(2).Info("create volume", "name", name, "http", httpRes.Status)
+
+	status := &common.ResponseStatus{}
+	if response != nil && len(response.GetStatus()) > 0 {
+		status = CreateCommonStatusFromStatus(&response.Status[0])
+	}
+
+	volume := common.VolumeObject{}
+	if response != nil && len(response.GetVolumes()) > 0 {
+		v := response.Volumes[0]
+		volume.ObjectName = v.GetObjectName()
```

Comment:
Copying every field manually seems like it will lead to bugs over time--at least, it seems like there should be a separate function so you don't have to repeat all this code each time.